### PR TITLE
#216 Update project configuration cli interface

### PR DESCRIPTION
**Pull request by slothspot**
**READY FOR REVIEW**

### CODE REVIEW

File: cli/pazuzu/actions/project.go [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+	pazuzufilePath := utils.GetAbsoluteFilePath(destination, pazuzu.PazuzufileName)
+	pazuzuFile, success := utils.ReadPazuzuFile(pazuzufilePath)
+
+	if success {
+		base = pazuzuFile.Base
+		features = pazuzuFile.Features
+	}
+
+	if key != "base" {
```
Comment (fehmicansaglam) [inline cli/pazuzu/actions/project.go]: What if we used `switch` here? IMHO it would be easier to add new keys this way.

```go
switch key {
case "base": 
       // ...
default: 
        return errors.New("Key is not supported")
}
```

**Comment by fehmicansaglam** [comment]
And also I suppose you need to update the README for the new layout unfortunately :)

File: cli/pazuzu/actions/project.go (hunk 1 above)
Comment (slothspot) [inline cli/pazuzu/actions/project.go]: Support of other keys would require change of PazuzuFile structure, so I propose to make this code change in separate commit when new configuration features would be added to PazuzuFile

**Comment by fehmicansaglam** [comment]
👍

File: cli/pazuzu/actions/config.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	"github.com/urfave/cli"
+	"github.com/zalando-incubator/pazuzu"
+	"github.com/zalando-incubator/pazuzu/config"
+	"os"
+	"text/tabwriter"
+)
+
+func ConfigShow(c *cli.Context) error {
+	if c.NArg() != 0 {
+		return errors.New("Wrong number of arguments")
```
Comment (erthalion) [inline cli/pazuzu/actions/config.go]: Maybe it makes sense to introduce few custom self-describing error types? 
Just to separate error representation, something like:

```go
if c.NArg() != 0 {
    return WrongArgumentsNumber()
}
```

Comment (slothspot) [inline cli/pazuzu/actions/config.go]: I agree on this. Should I do this within this commit or in separate one?

Comment (erthalion) [inline cli/pazuzu/actions/config.go]: I think in separate one, since it's not related to the original PR purpose. What do you think?

Comment (slothspot) [inline cli/pazuzu/actions/config.go]: Ok, I'll do it in separate and will check other possible places to generalize errors generation.

File: cli/pazuzu/actions/search.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	"regexp"
+	"text/tabwriter"
+)
+
+func Search(c *cli.Context) error {
+	if c.NArg() != 1 {
+		return errors.New("Search regexp is not provided")
+	}
+	feature := c.Args().First()
+	featureRegexp, err := regexp.Compile(feature)
```
Comment (erthalion) [inline cli/pazuzu/actions/search.go]: I see that pattern a lot in here. I understand, that it's how anyone uses golang, but maybe we can wrap it a bit, like this:

```go
featureRegexp := get_value(regepx.Compile(feature), BadRegexp)
cfg := config.GetConfig()
storage := get_value(config.GetStorageReader(*cfg), BadStorageReader)
features := get_value(storage.SearchMeta(featureRegexp), CantSearch)
``` 
and
```go
func get_value(value interface{}, err interface{}, errorType interface{}) {
    if err != nil {
        return errorType()
    } else {
        return value
    }
}
```

What do you think?

Comment (erthalion) [inline cli/pazuzu/actions/search.go]: On the second thought there will be a lot of type conversion when we get value back from this function.

**Comment by erthalion** [comment]
:+1:
